### PR TITLE
🔓 Add lock details and proper balance calculations

### PR DESCRIPTION
**Pull request by DevNikita16**
I've done proper balance calculation for total and transferable according to Bedeho's message.
And also basic locks details were added in this PR.

See #303

### CODE REVIEW

File: packages/ui/src/app/pages/Profile/components/AccountItem.tsx [hunk 1]
```diff
@@ -36,7 +38,9 @@ export const AccountItem = ({ account }: AccountItemDataProps) => {
           <DropDownButton onClick={setDropped} isDropped={isDropped} size="medium" />
         </AccountControls>
       </AccountItemWrap>
-      <DropDownToggle isDropped={isDropped}>I'm opened</DropDownToggle>
+      <DropDownToggle isDropped={isDropped}>
+        <BalanceDetails locks={balance?.locks || []} recoverables={[]} />
+      </DropDownToggle>
```
Comment (jodator) [inline packages/ui/src/app/pages/Profile/components/AccountItem.tsx:43]: Just because we have proper slices of the UI part on the way please revert the changes for the opened part.

**Review by jodator — CHANGES_REQUESTED** [review]
To properly finish this one:

- the locks details must wait for the UI. Right now you can display only a small icon indicator for the lock.
- because we don't have the data yet, please add a storybook story that can display this component with various locks (Joystream defines [almost 10 of them](https://joystream.gitbook.io/joystream-handbook/key-concepts/staking#locks-and-binding)).
- the icons can be anything (any other existing icon) for now, we'll add proper icons later.

File: packages/ui/src/app/pages/Profile/components/AccountItem.tsx [hunk 2]
```diff
@@ -163,12 +171,32 @@ export const AccountItem = ({ account }: AccountItemDataProps) => {
...
   return (
     <>
       <AccountItemWrap key={address}>
         <AccountInfo account={account} />
         <TokenValue value={balance?.total} />
-        <TokenValue value={balance?.locked} />
+        <div>
+          <TokenValue value={balance?.locked} />
+          <br />
+          {displayLockIcons()}
```
Comment (jodator) [inline packages/ui/src/app/pages/Profile/components/AccountItem.tsx:198]: Can you extract a component for this? I see that we might display account locks elsewhere as well so it will be helpful.

Also please create a storybook story for this new component that will display available locks.

**Review by jodator — CHANGES_REQUESTED** [review]
Thanks for the PR :) I see some room for improvement:

- having locks map and fluent locks ids
- separate component for displaying locks

File: packages/ui/src/app/pages/Profile/components/AccountItem.tsx (hunk 2 above)
Comment (DevNikita16) [reply packages/ui/src/app/pages/Profile/components/AccountItem.tsx]: ✅

File: packages/ui/src/accounts/types/balances.ts [hunk 3]
```diff
@@ -9,6 +15,10 @@ export interface Balances {
   transferable: BN
 }
 
+export interface DetailedBalances extends Balances {
+  locks: BalanceLock[]
+}
```
[the remark this answers is not in the document]
Comment (DevNikita16) [reply packages/ui/src/accounts/types/balances.ts]: ✅

File: packages/ui/src/accounts/hooks/useBalance.ts [hunk 4]
```diff
@@ -1,23 +1,45 @@
...
+import { lockTypes } from '@/accounts/model/lockTypes'
+import { useApi } from '@/common/hooks/useApi'
+import { useObservable } from '@/common/hooks/useObservable'
+import { Address } from '@/common/types'
 
-export function toBalances(balances: DeriveBalancesAll): Balances {
+import { DetailedBalances } from '../types'
+
+function lockLookup(id: LockIdentifier) {
+  const foundLock = lockTypes.find((lockType) => id.eq(lockType.type))
```
[the remark this answers is not in the document]
Comment (DevNikita16) [reply packages/ui/src/accounts/hooks/useBalance.ts]: ✅

File: packages/ui/src/app/pages/Profile/components/AccountItem.tsx [hunk 5]
```diff
@@ -19,26 +19,34 @@ import { Sizes, Colors } from '@/common/constants'
...
-  account: Account
+  lock: BalanceLock
 }
 
-export const DetailsItem = ({ account }: DetailsItemDataProps) => {
-  const address = account.address
-  const balance = useBalance(address)
+export const lockIcon = (id: number) => {
+  switch (id) {
+    case 10:
```
[the remark this answers is not in the document]
Comment (DevNikita16) [reply packages/ui/src/app/pages/Profile/components/AccountItem.tsx]: ✅
